src/cli/cmd_build_and_check.rs

```diff
@@ -116,6 +130,9 @@ pub fn cmd_build(args: CmdBuild) -> miette::Result<()> {
 
     println!("successfully built");
 
+    // server mode
+    handle.map(|h| h.join().unwrap());
```

Comment:
Since we don't use the return from the map function, maybe be better to this instead:

```suggestion
    if let Some(h) = handle { h.join().unwrap() }
```

Reply:
what's the difference?

Reply:
then clippy won't throw warning :)